### PR TITLE
feature(k8s-ingress): tag load ingress balancers

### PR DESCRIPTION
**Pull request by fruch**
since we are now creating the new AWS resources
we should tag them correctly for cleanups and billing

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
Still in draft since we need to setup permission for using the tagging api (works for my user, but not for all of dev users)
```
01:08:52  botocore.exceptions.ClientError: An error occurred (AccessDeniedException) when calling the GetResources operation: User: arn:aws:iam::***:user/** is not authorized to perform: tag:GetResources because no identity-based policy allows the tag:GetResources action
```

File: sdcm/k8s_configs/ingress-controller/10_haproxy-kubernetes-ingress.svc.yaml [hunk 1]
```diff
@@ -3,6 +3,9 @@ kind: Service
 metadata:
   name: haproxy-kubernetes-ingress
   namespace: haproxy-controller
+  annotations:
```
Comment (vponomaryov) [inline sdcm/k8s_configs/ingress-controller/10_haproxy-kubernetes-ingress.svc.yaml:6]: Adding the link to the docs you mention in the task for this PR would be very useful:

https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/service/annotations/#aws-resource-tags

File: sdcm/utils/common.py [hunk 2]
```diff
@@ -911,6 +912,81 @@ def clean_test_security_groups(tags_dict, dry_run=False):
...
+    :param dry_run: if True, wouldn't delete any resource
+    :return: None
+    """
+
+    # don't if `post_behavior_k8s_cluster` was set to not destroy
+    if "NodeType" in tags_dict and tags_dict.get("NodeType") != "k8s":
+        return
+
+    assert tags_dict, "tags_dict not provided (can't clean all instances)"
+    aws_instances = list_load_balancers_aws(tags_dict=tags_dict, group_as_region=True)
```
Comment (vponomaryov) [inline sdcm/utils/common.py:972]: aws_instances?

File: sdcm/utils/common.py [hunk 3]
```diff
@@ -911,6 +912,81 @@ def clean_test_security_groups(tags_dict, dry_run=False):
...
+            LOGGER.info("There are no ELBs to remove in AWS region %s", region)
+            continue
+        client = boto3.client('elb', region_name=region)
+        for elb in elb_list:
+            arn = elb.get('ResourceARN')
+            LOGGER.info("Going to delete '%s'", arn)
+            if not dry_run:
+                try:
+                    response = client.delete_load_balancer(LoadBalancerName=arn.split('/')[1])
+                    LOGGER.debug("Done. Result: %s\n", response)
```
Comment (vponomaryov) [inline sdcm/utils/common.py:985]: Maybe better to make all such calls be parallel/concurrent?
None of load balancers depends on any other.

Comment (fruch) [reply sdcm/utils/common.py:985]: since we have only one per test (for now, later we'll have one per region), it's doesn't make a lot of difference.

Comment (vponomaryov) [reply sdcm/utils/common.py:985]: How fast is it? If seconds, then ok.

Comment (fruch) [reply sdcm/utils/common.py:985]: we I didn't delete a lot of those, main one or two, and yest it was quick (it's not like deleting a node pool or eks cluster)

File: sdcm/k8s_configs/ingress-controller/10_haproxy-kubernetes-ingress.svc.yaml (hunk 1 above)
Comment (fruch) [reply sdcm/k8s_configs/ingress-controller/10_haproxy-kubernetes-ingress.svc.yaml:6]: added

**Pull request merged**